module/mutect2.nf

```diff
@@ -90,12 +92,54 @@ workflow mutect2 {
             run_MergeMutectStats_GATK.out.merged_stats,
             run_LearnReadOrientationModel_GATK.out.read_orientation_model,
             contamination_table.collect()
+            )
+        filter_VCF_BCFtools(run_FilterMutectCalls_GATK.out.filtered
+            .map{ it -> ['all', it] }
+            )
+        split_VCF_BCFtools(filter_VCF_BCFtools.out.gzvcf
+            .map{ it -> it[1] },
+            ['snps', 'mnps', 'indels']
         )
-        filter_VCF_BCFtools(run_FilterMutectCalls_GATK.out.filtered.map{ it -> ['all', it] })
-        split_VCF_BCFtools(filter_VCF_BCFtools.out.gzvcf.map{ it -> it[1] }, ['snps', 'mnps', 'indels'])
-        compress_index_VCF(split_VCF_BCFtools.out.gzvcf)
-        file_for_sha512 = compress_index_VCF.out.index_out.map{ it -> ["mutect2-${it[0]}-vcf", it[1]] }
-            .mix( compress_index_VCF.out.index_out.map{ it -> ["mutect2-${it[0]}-index", it[2]] } )
+
+        if (params.tumor_only_mode) {
+            old_normal_id = 'NO_NAME'
+        } else {
+        old_normal_id = params.samples_to_process
+            .findAll { it['sample_type'] == 'normal' }
+            .get(0)['orig-id']
+        }
+        old_tumor_id = params.samples_to_process
+            .findAll { it['sample_type'] == 'tumor' }
+            .get(0)['orig-id']
+
+        split_VCF_BCFtools.out.gzvcf
+            .map { it -> [it, old_normal_id, old_tumor_id] }
+            .map { it -> [it[1], it[2]] }
+            .set { old_names_ch}
+
+        if (params.single_NT_paired) {
+            split_VCF_BCFtools.out.gzvcf
+                .map { it -> [it, params.normal_id, params.tumor_id] }
+                .map { it -> [it[1], it[2]] }
+                .set { new_names_ch }
+        } else {
+            // this does not lead to any id changes and is needed to correctly track filenames
+            split_VCF_BCFtools.out.gzvcf
+                .map { it -> [it, old_normal_id, old_tumor_id] }
+                .map { it -> [it[1], it[2]] }
+                .set { new_names_ch }
+        }
```

Comment:
So for tumor-only, normal-only, multi-normal, and multi-tumor modes, the VCF headers aren't being updated with the sanitized IDs? Any reason for that since there would now be a difference in the behavior and expected sample IDs that's purely dependent on the mode run with call-sSNV rather than a standard expected behavior with the pipeline in general (all IDs will always be sanitized)

Reply:
done